04-onion-routing.md

```diff
@@ -257,6 +257,9 @@ This is a more flexible format, which avoids the redundant `short_channel_id` fi
     1. type: 6 (`short_channel_id`)
     2. data:
         * [`short_channel_id`:`short_channel_id`]
+    1. type: 8 (`basic_mpp`)
```

Comment:
Indentation and numbering seem strange to me.

Reply:
Even types are used because those are mandatory fields, does that help?

Reply:
Numbering I am referring to is before `.` . I.e. `1. type` then `2. data` then `1. type`.

Also next line is more indent but starts at `2.`

Reply:
Yes, these are a list of separate TLV record types.

Reply:
Seems we should change the formatting somehow?  If you format like that, this is what Markdown will produce:

> 1. type
> 2. data
> 1. type
>    2. data

Reply:
Ah, yuck.  I missed that it misrenders.  Let me see if I can fix that somehow :(

Reply:
OK, I think the neatest change is to change them to '*' and fix the tooling to allow that:

```markdown
1. tlvs: `tlv_payload`
2. types:
    * type: 0 (`destination_signal`)
    * data:
	    * Empty
    * type: 2 (`amt_to_forward`)
    * data:
        * [`tu64`:`amt_to_forward`]
    * type: 4 (`outgoing_cltv_value`)
    * data:
        * [`tu32`:`outgoing_cltv_value`]
    * type: 6 (`short_channel_id`)
    * data:
        * [`short_channel_id`:`short_channel_id`]
    * type: 8 (`basic_mpp`)
	* data:
	    * [`tu64`:`total_msat`]
```

This needs to be tweaked to parse anyway, let's wait until #631 which fixes the tooling, then I'll add a fixup for this...